packages/block-editor/src/components/global-styles/get-global-styles-changes.js

```diff
@@ -176,29 +171,75 @@ function getGlobalStylesChangelist( next, previous ) {
 }
 
 /**
- * From a getGlobalStylesChangelist() result, returns a truncated array of translated changes.
- * Appends a translated string indicating the number of changes that were truncated.
+ * From a getGlobalStylesChangelist() result, returns an array of translated global styles changes, grouped by type.
+ * The types are 'blocks', 'elements', 'settings', and 'styles'.
  *
  * @param {Object}              next     The changed object to compare.
  * @param {Object}              previous The original object to compare against.
  * @param {{maxResults:number}} options  Options. maxResults: results to return before truncating.
- * @return {string[]}                        An array of translated changes.
+ * @return {string[]}                    An array of translated changes.
  */
 export default function getGlobalStylesChanges( next, previous, options = {} ) {
-	const changes = getGlobalStylesChangelist( next, previous );
-	const changesLength = changes.length;
+	let changeList = getGlobalStylesChangelist( next, previous );
+	const changesLength = changeList.length;
 	const { maxResults } = options;
 
-	// Truncate to `n` results if necessary.
-	if ( !! maxResults && changesLength && changesLength > maxResults ) {
-		const deleteCount = changesLength - maxResults;
-		const andMoreText = sprintf(
-			// translators: %d: number of global styles changes that are not displayed in the UI.
-			_n( '…and %d more change', '…and %d more changes', deleteCount ),
-			deleteCount
-		);
-		changes.splice( maxResults, deleteCount, andMoreText );
+	if ( changesLength ) {
+		// Truncate to `n` results if necessary.
+		if ( !! maxResults && changesLength > maxResults ) {
+			changeList = changeList.slice( 0, maxResults );
+		}
+		return Object.entries(
+			changeList.reduce( ( acc, curr ) => {
+				const group = acc[ curr[ 0 ] ] || [];
+				if ( ! group.includes( curr[ 1 ] ) ) {
+					acc[ curr[ 0 ] ] = [ ...group, curr[ 1 ] ];
+				}
+				return acc;
+			}, {} )
+		).map( ( [ key, changeValues ] ) => {
+			const changeValuesLength = changeValues.length;
+			switch ( key ) {
+				case 'blocks': {
+					return sprintf(
+						// translators: %2$s: a list of block names separated by a comma.
+						_n( '%2$s block.', '%2$s blocks.', changeValuesLength ),
```

Comment:
```suggestion
						_n( '%s block.', '%s blocks.', changeValuesLength ),
```

Reply:
This is not quite right I believe.

 It will output `5 blocks.`

With `_n` we need to pass the integer (for the plural) - this is first argument of `__sprintf` (`%1$s`) so we need the second `%2$s`

Reply:
I see, then we  should be able to omit the unused first argument of `sprintf`.

```javascript
return sprintf(
	// translators: %s: a list of block names separated by a comma.
	_n( '%s block.', '%s blocks.', changeValuesLength ),
	joinedChangesValue
);
```

However, the current implementation should work fine 👍

Reply:
@t-hamano Sorry! 🤦🏻 I'll read more closely next time.

You're right. I've updated this over here:

- https://github.com/WordPress/gutenberg/pull/59160

Thank you!